integration-tests/api/__tests__/store/orders.js

```diff
@@ -155,6 +155,50 @@ describe("/store/carts", () => {
       )
     })
 
+    it("lookup order response contains only fields defined with `fields` param", async () => {
+      const api = useApi()
+
+      const response = await api
+        .get(
+          "/store/orders?display_id=111&email=test@email.com&fields=status,object"
+        )
+        .catch((err) => {
+          return err.response
+        })
+
+      expect(response.data.order).toEqual(
+        expect.objectContaining({
+          object: "order",
+          status: expect.any(String),
+          items: expect.any(Array), // default relations are not filtered out
+        })
+      )
+
+      expect(response.data.order).not.toHaveProperty("id")
+      expect(response.data.order).not.toHaveProperty("created_at")
+    })
+
+    it("get order response contains only fields defined with `fields` param", async () => {
+      const api = useApi()
+
+      const response = await api
+        .get("/store/orders/order_test?fields=status,object")
+        .catch((err) => {
+          return err.response
+        })
+
+      expect(response.data.order).toEqual(
+        expect.objectContaining({
+          object: "order",
+          status: expect.any(String),
+          items: expect.any(Array), // default relations are not filtered out
+        })
+      )
+
+      expect(response.data.order).not.toHaveProperty("id")
```

Comment:
Suggestion: instead we can just expect { handle, variant: any, otherExpectedFields }. Here we could have other unexpected field that are not checked. If you agree, the other places need to be changed as well